fe/fe-core/src/main/java/com/starrocks/http/rest/TransactionLoadAction.java

```diff
@@ -184,192 +213,593 @@ public void executeWithoutPassword(BaseRequest request, BaseResponse response) t
+                if (TXN_BEGIN.equals(op)) {
+                    nodeId = GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo().seqChooseBackendOrComputeId();
                     // txnNodeMap is LRU cache, it atomic remove unused entry
-                    txnNodeMap.put(label, nodeID);
-                } else if (channelIdStr == null) {
-                    nodeID = txnNodeMap.get(label);
+                    txnNodeMap.put(label, nodeId);
+                } else {
+                    nodeId = txnNodeMap.get(label);
                 }
             }
+
+            if (nodeId == null) {
+                throw new UserException(String.format(
+                        "Transaction with op[%s] and label[%s] has no node.", op.getValue(), label));
+            }
+
+            return nodeId;
         }
 
-        if (op.equalsIgnoreCase(TXN_BEGIN) && channelIdStr != null) {
-            TransactionResult resp = new TransactionResult();
-            long timeoutMillis = 20000;
-            if (timeout != null) {
-                timeoutMillis = Long.parseLong(timeout) * 1000;
+    }
+
+    /* helper classes */
+
+    private static class OpMetrics {
+        LongCounterMetric opRunningNum;
+        Histogram opLatencyMs;
+
+        static OpMetrics of(LongCounterMetric opRunningNum, Histogram opLatencyMs) {
+            OpMetrics metrics = new OpMetrics();
+            metrics.opRunningNum = opRunningNum;
+            metrics.opLatencyMs = opLatencyMs;
+            return metrics;
+        }
+    }
+
+    /**
+     * Request params parser, validator and holder.
+     */
+    private static class RequestParams {
+
+        /* headers */
+
+        private final String dbName;
+        private final String tableName;
+        private final String label;
+        private final Operation operation;
+        private final Long timeoutMillis;
+        private final Channel channel;
+
+        /* queries */
+
+        private final LoadJobSourceType sourceType;
+
+        /* body */
+
+        private final Body body;
+
+        public RequestParams(String dbName,
+                             String tableName,
+                             String label,
+                             Operation operation,
+                             Long timeoutMillis,
+                             Channel channel,
+                             LoadJobSourceType sourceType,
+                             Body body) {
+            this.dbName = dbName;
+            this.tableName = tableName;
+            this.label = label;
+            this.operation = operation;
+            this.timeoutMillis = timeoutMillis;
+            this.channel = channel;
+            this.sourceType = sourceType;
+            this.body = body;
+        }
+
+        public static RequestParams from(BaseRequest request) throws UserException {
+            String dbName = request.getRequest().headers().get(DB_KEY);
+            if (StringUtils.isBlank(dbName)) {
```

Comment:
may be null

Reply:
When `dbName` is null, empty or blank, the `isBlank` will return true, you mean `dbName` is allowed to be null?